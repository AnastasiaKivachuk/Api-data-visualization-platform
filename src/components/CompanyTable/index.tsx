import React from "react";

import {Paper, Table, TableBody, TableCell, TableContainer, TableRow} from "@mui/material";

import {CompanyTableConfigInterface} from "../../interfaces/home.interface";

type Props = {
    config: CompanyTableConfigInterface[]
}

const CompanyTable = ({config}: Props): JSX.Element => {
    return (
        <TableContainer sx={{width: 'fit-content', margin: '20px auto'}} component={Paper}>
            <Table sx={{width: 450}} aria-label="simple table" align={'center'}>
                <TableBody>
                    {config.map(({leftColumn, rightColumn, id}) =>
                        <TableRow key={id}
                                  sx={{'&:last-child td, &:last-child th': {border: 0}}}
                        >
                            <TableCell component="th" scope="row">
                                {leftColumn}
                            </TableCell>
                            <TableCell align="right">
                                {rightColumn}
                            </TableCell>
                        </TableRow>)}
                </TableBody>
            </Table>
        </TableContainer>
    )
}

export default CompanyTable;
