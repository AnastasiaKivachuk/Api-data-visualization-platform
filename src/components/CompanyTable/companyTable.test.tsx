import {shallow} from "enzyme";
import Link from "next/link";
import React from "react";

import CompanyTable from "./index";

const link = 'https://twitter.com/elonmusk';

const MockConfig = [{
    id: 1,
    leftColumn: 'CEO',
    rightColumn: <Link href={'https://twitter.com/elonmusk'} target={'_blank'}>
        <a>
            Elon Musk
        </a>
    </Link>
},
    {
        id: 2,
        leftColumn: 'Employees',
        rightColumn: 7000
    },
]

const component = shallow(<CompanyTable config={MockConfig}/>);

describe("CompanyTable component", () => {

    it("should render CompanyTable component", () => {
        expect(component).toMatchSnapshot();
    })

    it("check all elements", () => {
        expect(component.find('ForwardRef(TableRow)').length).toBe(2)
        expect(component.find('ForwardRef(TableCell)').length).toBe(4)
    })

    it("check all links", () => {
        expect(component.find('a').length).toBe(1)
        expect(component.find('ForwardRef').prop('href')).toBe(link)
    })
})
