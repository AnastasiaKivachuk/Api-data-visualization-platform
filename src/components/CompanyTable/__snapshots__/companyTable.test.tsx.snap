// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyTable component should render CompanyTable component 1`] = `
<ForwardRef(TableContainer)
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "children": [Function],
        "className": [Function],
        "classes": [Function],
        "component": [Function],
        "elevation": [Function],
        "square": [Function],
        "sx": [Function],
        "variant": [Function],
      },
      "render": [Function],
    }
  }
  sx={
    Object {
      "margin": "20px auto",
      "width": "fit-content",
    }
  }
>
  <ForwardRef(Table)
    align="center"
    aria-label="simple table"
    sx={
      Object {
        "width": 450,
      }
    }
  >
    <ForwardRef(TableBody)>
      <ForwardRef(TableRow)
        key="1"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          CEO
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="right"
        >
          <ForwardRef
            href="https://twitter.com/elonmusk"
            target="_blank"
          >
            <a>
              Elon Musk
            </a>
          </ForwardRef>
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="2"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Employees
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="right"
        >
          7000
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
    </ForwardRef(TableBody)>
  </ForwardRef(Table)>
</ForwardRef(TableContainer)>
`;
